walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -51,43 +53,41 @@
 // Group together the leftover time ranges for results
     for(TimeRange range : initalRanges) {
 
-        //range is not schedulable and there was a previous valid range started
-        if(!range.isSchedulable() && tempRange.isValidRange()) {
-            if(tempRange.getDuration() >= duration) {
-            results.add(TimeRange.fromStartDuration(tempRange.getStart(), tempRange.getDuration()));
-            }
-            tempRange.setValidRange(false);
-            tempRange.setStart(0);
-            tempRange.setDuration(0);
-        }
-
-        //range is schedulable and there is a valid range in the works
-        if(range.isSchedulable() && tempRange.isValidRange()) {
-            tempRange.setDuration(tempRange.getDuration() + range.getDuration());
-        }
-
-        //range is schedulable and there is not a vaild range already in the works then start a valid range
-        if(range.isSchedulable() && !tempRange.isValidRange()) {
-            tempRange.setValidRange(true);
-            tempRange.setStart(range.getStart());
-            tempRange.setDuration(range.getDuration());
-        }
+      //if range isn't valid then add tempRange to results then reset it
+      if(!range.hasPositiveDuration()) {
+        maybeAddToResults(tempRange, results, duration);
+        tempRange.setStart(0);
+        tempRange.setDuration(0);
+      }
 
-        //If it is the last range of the initial ranges and there is a viable range in the works and
-        //the range duration statisfies the required duration then add it to results
-        if(range.end() == TimeRange.END_OF_DAY && tempRange.isValidRange() && tempRange.getDuration() >= duration) {
-            results.add(TimeRange.fromStartEnd(tempRange.getStart(), TimeRange.END_OF_DAY, true));
-        }
+      //if range is valid then start a new tempRange if there was not one already
+      if(!tempRange.hasPositiveDuration() && range.hasPositiveDuration()) {
```

Comment:
if you continue above, you can assume range will always have a positive duration